src/slider/slider.js

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {thumbWidth} from './constants';
+import type {StatelessStateT, PropsT} from './types';
+import {
+  Root as StyledRoot,
+  Axis as StyledAxis,
+  AxisRange as StyledAxisRange,
+  Tick as StyledTick,
+  TickBar as StyledTickBar,
+  Thumb as StyledThumb,
+} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+/* global document */
+/* global window */
+
+class Slider extends React.Component<PropsT, StatelessStateT> {
+  domNode: ?Element | Text;
+  onMouseUpListener: ?MouseEventHandler;
+  onMouseMoveListener: ?MouseEventHandler;
+  static defaultProps = {
+    overrides: {},
+    onChange: () => {},
+    onAxisClick: () => {},
+    onMouseDown: () => {},
+    onMouseMove: () => {},
+    onMouseUp: () => {},
+    error: false,
+    autoFocus: false,
+    tabIndex: 0,
+  };
+
+  state = {
+    isThumbMoving: false,
+    currentThumb: -1,
+    currentMove: 0,
+    thumbRefs: this.props.value.map(() => React.createRef()),
+  };
+
+  componentDidMount() {
+    /* eslint-disable-next-line react/no-find-dom-node */
+    this.domNode = findDOMNode(this);
+    this.addDocumentMouseEvents();
+  }
+
+  componentWillUnmount() {
+    this.removeDocumentEvents();
+  }
+
+  onMouseDown = (event: MouseEvent, thumbIndex: number) => {
+    this.setState(
+      {
+        isThumbMoving: true,
+        currentThumb: thumbIndex,
+      },
+      () =>
+        this.props.onMouseDown({
+          event,
+          currentThumb: thumbIndex,
+          isThumbMoving: true,
+        }),
+    );
+  };
+
+  onMouseUp = (event: MouseEvent) => {
+    const {isThumbMoving} = this.state;
+    if (isThumbMoving) {
+      this.setState(
+        {
+          isThumbMoving: false,
+          currentThumb: -1,
+          currentMove: 0,
+        },
+        () =>
+          this.props.onMouseUp({
```

Comment:
if we need to pass `isThumbMoving` value, than we need to wait for it to be switched after `setState`. Otherwise if call immediately, the value will be inconsistent.